pinc/unicode.inc

```diff
@@ -61,27 +87,31 @@ function utf8_filter_out_codepoints($string, $remove_codepoints, $replacement=""
     return preg_replace($pattern_string, $replacement, $string);
 }
 
-# Take an array of Unicode codepoints (U+####), or codepoint ranges
-# (U+####-U+####) and return an array of unicode characters.
+# Take an array of Unicode codepoints (U+####), codepoint ranges
+# (U+####-U+####), or combined characters (U+####>U+####) and
+# return an array of unicode characters.
 function convert_codepoint_ranges_to_characters($codepoints)
 {
     $return_array = [];
     if(!is_array($codepoints))
         $codepoints = [$codepoints];
+
     foreach($codepoints as $codepoint)
     {
         if($codepoint === NULL)
         {
             $return_array[] = NULL;
         }
-        elseif(strpos($codepoint, '-') === False)
+        if(strpos($codepoint, '-') !== False)
         {
-            $return_array[] = utf8_chr($codepoint);
+            # we have a character range
+            list($start, $end) = explode('-', $codepoint);
+            $return_array = array_merge($return_array, utf8_range($start, $end));
         }
         else
         {
-            list($start, $end) = explode('-', $codepoint);
-            $return_array = array_merge($return_array, utf8_range($start, $end));
+            # a single character or a combined character
+            $return_array[] = utf8_combined_chr($codepoint);
```

Comment:
Need to deal with single codepoints separately since the  version of utf8_combined_chr() in this file doesn't check for single characters.

Reply:
That's the beautify of your `utf8_combined_chr()` version -- it does!